### PR TITLE
Fix dependencies of mbedtls_pk_ec_ro and mbedtls_pk_ec_rw

### LINKED ISSUE #8263: Fix dependencies of mbedtls_pk_ec_ro and mbedtls_pk_ec_rw
> In pk_internal, I think the new functions mbedtls_pk_ec_ro and mbedtls_pk_ec_rw should depend on !MBEDTLS_PK_USE_PSA_EC_DATA as they don't really make sense with MBEDTLS_PK_USE_PSA_EC_DATA. — @mpg

Split out of https://github.com/Mbed-TLS/mbedtls/issues/7757.

### PR DESCRIPTION
**Pull request by valeriosetti**
## Description

This PR resolves #8263 by using `!MBEDTLS_PK_USE_PSA_EC_DATA` as guard for `mbedtls_pk_ec_[ro/rw]` and fixing related issues.

## PR checklist

- [ ] **changelog** not required as nothing changes for the end user
- [ ] **backport** not required
- [ ] **tests** not required as no new code is introduced and this part is already tested by existing ones

### CODE REVIEW

File: library/ssl_tls12_server.c [hunk 1]
```diff
@@ -676,7 +676,11 @@ static int ssl_check_key_curve(mbedtls_pk_context *pk,
                                uint16_t *curves_tls_id)
 {
     uint16_t *curr_tls_id = curves_tls_id;
+#if defined(MBEDTLS_PK_USE_PSA_EC_DATA)
```
Comment (mpg) [inline library/ssl_tls12_server.c:679]: Suggestion: use `mbedtls_pk_get_group_id()`.

**Review by mpg — COMMENTED** [review]
LGTM, but I suggested a small improvement.

File: library/ssl_tls12_server.c (hunk 1 above)
Comment (valeriosetti) [reply library/ssl_tls12_server.c]: Nice! I added that function but I totally forgot about it :D

**Review by mpg — APPROVED** [review]
LGTM

**Review by gabor-mezei-arm — APPROVED** [review]
LGTM

**Pull request merged**